lang/src/parser.rs

```diff
@@ -4,71 +4,169 @@
 //! Unlike most ASTs, CSTs preserve comments and whitespace. Additionally,
 //! Rowan particularly does not distinguish trivia from regular nodes, and uses
 //! the same enum for both leaf nodes (tokens) and composite nodes (trees).
+//!
+//! This module is "glue code" between rowan and chumsky.
+//! The actual parser lives in `grammar.rs`.
 
 // separate mod to encapsulate the unsafety
 mod syntax {
-    #[repr(u16)]
+    use cstree::RawSyntaxKind;
+
+    #[repr(u32)]
     #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    #[allow(non_camel_case_types)]
     pub enum SyntaxKind {
         // leaf nodes
+        // symbols
         NEWLINE = 0,
-        WHITESPACE,
         CELL,
         EQ,
         INT,
+        COLON,
+        DOLLAR,
+        // LEFT_BRACKET,
+        // RIGHT_BRACKET,
+
+        // keywords
+        ALIAS_TOK,
+        ENUM_TOK,
+
+        // data tokens
+        WHITESPACE,
+        COMMENT,
+        IDENT,
+        STR,
 
         // composite nodes
+
+        // expressions
+        CELL_RANGE,
+        PLACE,
+        ALIAS_EXPR,
+        ENUM_EXPR,
+        EXPR,
+
+        // statements
         ASSIGN,
+        ALIAS_STMT,
         STATEMENT,
+        // ARRAY_RANGE,
+
         // this MUST come last in the enum; we depend on it for memory safety
         ROOT,
     }
 
-    impl From<SyntaxKind> for rowan::SyntaxKind {
+    impl From<SyntaxKind> for RawSyntaxKind {
         fn from(kind: SyntaxKind) -> Self {
-            Self(kind as u16)
+            Self(kind as u32)
         }
     }
 
-    #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
-    pub enum Lang {}
-    impl rowan::Language for Lang {
-        type Kind = SyntaxKind;
-        fn kind_from_raw(raw: rowan::SyntaxKind) -> Self::Kind {
-            assert!(raw.0 <= Self::Kind::ROOT as u16);
+    impl cstree::Syntax for SyntaxKind {
+        fn from_raw(raw: RawSyntaxKind) -> Self {
+            assert!(raw.0 <= Self::ROOT as u32);
             // SAFETY: we just checked this is a valid variant.
-            unsafe { std::mem::transmute::<u16, SyntaxKind>(raw.0) }
+            unsafe { std::mem::transmute::<u32, SyntaxKind>(raw.0) }
```

Comment:
By the way, if you're not opposed to proc-macros, `cstree` has a `derive` feature with a derive macro for `Syntax` that automates this conversion, including the `unsafe` stuff.